ray_on_golem/provider/node_provider.py

```diff
@@ -290,6 +295,10 @@ def _apply_config_defaults(config: Dict[str, Any]) -> None:
 
         for node_type in config.get("available_node_types", {}).values():
             node_config = deepcopy(config["provider"]["parameters"]["node_config"])
+
+            if node_type == "ray.head.default":
+                dpath.merge(node_config, deepcopy(HEAD_NODE_DEFAULTS))
```

Comment:
@approxit doesn't this mean that the setting defined in `HEAD_NODE_DEFAULTS` will _always_ override whatever had been set in the node config in YAML?